secrethub/resource_access_rule.go

```diff
@@ -0,0 +1,84 @@
+package secrethub
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/secrethub/secrethub-go/internals/api"
+)
+
+func resourceAccessRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAccessRuleSet,
+		Read:   resourceAccessRuleRead,
+		Update: resourceAccessRuleSet,
+		Delete: resourceAccessRuleDelete,
+		Schema: map[string]*schema.Schema{
+			"dir_path": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The path of the directory on which the permission holds.",
+			},
+			"account_name": {
```

Comment:
Not completely sure what name to use here. For now it accepts account names (usernames or service IDs), but I can imagine we'll add setting access rules by account ID later.

Reply:
Maybe just `account`, and change the description to: `The username or service ID...`

Reply:
And when we add support for supplying IDs? Will we supply those on `account` as well? I believe we can only determine the difference between UUID and username based on the length. Are we comfortable setting these maxima as hard requirements, so we can keep this length check as a way to differentiate the two? Then we cannot relax these constraints later.

Reply:
Discussed this in-person. We also use `account_name` in the API for referring to both username and service ID. We'll also do so here.